### ISSUE TITLE
[new option]: "allow" option for quote-props to allow pattern

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13226]
Ref https://github.com/eslint/eslint/issues/13053 

**What rule do you want to change?**
`quote-props`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option 

**Please provide some example code that this change will affect:**

```js
/*eslint quote-props: ["error", "always", { allow: ['^_'] }]*/
var object = {
   _foo: "bar", // lint free
    baz: 42, // error
};
```

**What does the rule currently do for this code?**
throws error 

**What will the rule do after it's changed?**
ignores them

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by mdjermanovic** [comment]
Seems reasonable to me, given the use case described in #13053 and the fact that `dot-notation` already has a similar option.
